test/yaml/problem/invalid.yaml

```diff
@@ -22,7 +22,7 @@ yaml:
   limits:
     time_multipliers:
       mumbo: jumbo
-warn: "found unknown problem.yaml key: mumbo in `limits.time_multipliers`"
+warn: "found unknown problem.yaml key: mumbo in `time_multipliers`"
```

Comment:
I don't like that the warning no longer shows the full path. Many keys can occur in different paths in the YAML tree, e.g. keys for authors in `credits`

Reply:
We never did this for any other key? Removing the `limits.` here only made it consistent with every other place which allows reusing code (also this key is also globally unique).
So yeah in general I am fine with giving more info, but in most cases that is not necessary and with this PR I don't really want to change any behavior, just cleanup/type check the code better